pallets/order-book/src/lib.rs

```diff
@@ -847,36 +798,40 @@ pub mod pallet {
 			account: T::AccountId,
 			currency_in: T::AssetCurrencyId,
 			currency_out: T::AssetCurrencyId,
-			buy_amount: T::Balance,
-			sell_rate_limit: T::SellRatio,
-			min_fulfillment_amount: T::Balance,
-			validate: impl FnOnce(&OrderOf<T>) -> DispatchResult,
+			amount_out: T::Balance,
+			price: OrderPrice<T::SellRatio>,
+			min_amount_out: T::Balance,
 		) -> Result<T::OrderIdNonce, DispatchError> {
 			<OrderIdNonceStore<T>>::try_mutate(|n| {
 				*n = n.ensure_add(T::OrderIdNonce::one())?;
 				Ok::<_, DispatchError>(())
 			})?;
 
-			let max_sell_amount =
-				Self::convert_with_ratio(currency_in, currency_out, sell_rate_limit, buy_amount)?;
+			let order_id = <OrderIdNonceStore<T>>::get();
+
+			let min_fulfillment_amount_out = T::DecimalConverter::to_asset_balance(
+				T::MinFulfillmentAmountNative::get(),
```

Comment:
Wasn't there a discussion on slack regarding this? At least how I thing this is working, that should be fine, right?

Reply:
See the previous comment; it's working as before, and no changes did regarding this. I just move this call into the `inner_place_order()` function.

Reply:
Cool.